File: src/test/java/cn/edu/gxust/jiweihuang/java/math/function/univariate/QuadraticVertexLogisticTest.java
package cn.edu.gxust.jiweihuang.java.math.function.univariate;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;


class QuadraticVertexLogisticTest {
    private final QuadraticVertexLogistic qvl = new QuadraticVertexLogistic(0.5E-5, 1200, 0, 1, 0.005, 300, 1.8);

    @BeforeEach
    void setUp() {
        System.out.println("Test the class QuadraticVertexLogistic");
        System.out.println(qvl);
        System.out.println(qvl.formula());
    }

    @AfterEach
    void tearDown() {
        System.out.println("Test the class QuadraticVertexLogistic Over");
    }

    @Test
    void quadraticVertexA() {
        assertEquals(0.5e-5, qvl.getQuadraticVertexA());
    }

}

File: src/test/scala/cn/edu/gxust/jiweihuang/scala/math/function/univariate/QuadraticVertexLogisticTest.scala
package cn.edu.gxust.jiweihuang.scala.math.function.univariate

import cn.edu.gxust.jiweihuang.scala.test.UnitSpec

class QuadraticVertexLogisticTest extends UnitSpec {
  "" should "" in {
    println(QuadraticVertexLogistic(1e-2,1200,0,1,0.005,300,1.5).integrateRomberg(0, 1000))
    println(QuadraticVertexLogistic(1e-2,1200,0,1,0.005,300,1.5).integrateSimpson(0, 1000))
    println(QuadraticVertexLogistic(1e-2,1200,0,1,0.005,300,1.5).integrateTrapezoid(0, 1000))
    println(QuadraticVertexLogistic(1e-2,1200,0,1,0.005,300,1.5).integrateMidPoint(0, 1000))
    println(QuadraticVertexLogistic(1e-2,1200,0,1,0.005,300,1.5).integrateIterativeLegendreGauss(0, 1000))
  }

  "f" should "g" in {
    println(QuadraticVertexLogistic(3).differential(1))
    println(QuadraticVertexLogistic(3).derivative(1))
    println(QuadraticVertexLogistic(4, 5).differential(2))
    println(QuadraticVertexLogistic(4, 5).derivative(2))
    println(QuadraticVertexLogistic(4, 5, 6, 7).differential(3))
    println(QuadraticVertexLogistic(4, 5, 6, 7).derivative(3))
    println(QuadraticVertexLogistic(8, 9, 10, 11, 0, 2, 3).differential(4))
    println(QuadraticVertexLogistic(8, 9, 10, 11, 0, 2, 3).derivative(4))
  }
}
